### PR TITLE
[Feature][datasource] Change DataSource Connection Pool from Druid to HiKariCP (#6490)

### LINKED ISSUE #6490: [Feature] Change DataSource Connection Pool from Druid to HiKariCP
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Change DataSource Connection Pool from Druid to HiKariCP

### Use case

_No response_

### Related issues

#6448 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Narcasserun**
* Change DataSource Connection Pool from Druid to HiKariCP

This closes #6490

### CODE REVIEW

File: dolphinscheduler-dao/pom.xml [hunk 1]
```diff
@@ -106,10 +106,6 @@
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
         </dependency>
-        <dependency>
-            <groupId>com.alibaba</groupId>
```
Comment (kezhenxu94) [inline dolphinscheduler-dao/pom.xml:110]: Remove the license in all LICENSE file and known-dependencies.txt

Comment (Narcasserun) [reply dolphinscheduler-dao/pom.xml:110]: At present, dataX-task strongly relies on Druid for parsing. It is recommended to switch the metadata to provide interface parsing and then remove it

Comment (kezhenxu94) [reply dolphinscheduler-dao/pom.xml:110]: Got it

**Comment by zhongjiajie** [comment]
Hi @kezhenxu94 , could you please check why `e2e` failed? @Narcasserun ask me but I could not find some detail reason 😭 .

**Comment by Narcasserun** [comment]
6

**Review by kezhenxu94 — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**